universal-application-tool-0.0.1/test/app/SecurityBrowserTest.java

```diff
@@ -2,13 +2,50 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertTrue;
+import static play.test.Helpers.fakeApplication;
 
 import auth.Roles;
+import com.google.common.collect.ImmutableMap;
 import controllers.routes;
 import java.util.Optional;
+import org.junit.ClassRule;
 import org.junit.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+import play.Application;
+import support.TestConstants;
 
 public class SecurityBrowserTest extends BaseBrowserTest {
+  public static final DockerImageName OIDC_IMAGE =
+      DockerImageName.parse("public.ecr.aws/t1q6b4h2/oidc-provider:latest");
+
+  @ClassRule
+  public static GenericContainer<?> oidcProvider =
+      new GenericContainer<>(OIDC_IMAGE).withExposedPorts(3380);
+
+  @Override
+  protected Application provideApplication() {
+    ImmutableMap<String, Object> config =
+        new ImmutableMap.Builder<String, Object>()
+            .putAll(TestConstants.TEST_DATABASE_CONFIG)
+            .putAll(
+                TestConstants.oidcConfig(oidcProvider.getHost(), oidcProvider.getMappedPort(3380)))
+            .build();
+    return fakeApplication(config);
+  }
+
+  protected void loginWithSimulatedIdcs() {
```

Comment:
suggestion (non-blocking): maybe in another PR we could add something like this to the `BaseBrowserTest`?

Reply:
To do that we'd need the OIDC container, which is mildly expensive to start up, which is why I didn't do that ... if needed we'll move it around, though!  :)